proxy/src/coco/peer.rs

```diff
@@ -113,6 +125,67 @@ pub fn list_projects(peer: &PeerApi) -> Result<Vec<Project>, error::Error> {
         .collect()
 }
 
+/// Get all peer IDs and their branches.
+///
+/// # Errors
+///
+///   * [`error::Error::LibradLock`]
+///   * [`error::Error::Git`]
+pub fn remotes(
+    peer: &PeerApi,
+    owner: &User,
+    project_urn: &RadUrn,
+) -> Result<Vec<Remote>, error::Error> {
+    let project = get_project(peer, project_urn)?;
+    let peer_id = peer.peer_id();
+    let storage = peer.storage();
+    let repo = storage.open_repo(project.urn())?;
+    let refs = repo.rad_refs()?;
+
+    let (local_branches, local_tags) = with_browser(peer, &project.urn(), |browser| {
+        Ok((source::local_branches(browser)?, source::tags(browser)?))
+    })?;
+
+    let owner = owner.to_data().build()?; // TODO(finto): Dirty hack to make our Verified User into a Draft one
+    let mut remotes = vec![Remote {
+        user: owner,
+        branches: local_branches,
+        tags: local_tags,
+    }];
+
+    for remote in refs.remotes.flatten() {
+        let refs = if remote == peer_id {
+            // TODO(finto): Bug in librad
+            continue;
+        } else {
+            let remote_branches = storage.rad_refs_of(&project.urn(), remote.clone())?;
+            remote_branches
+                .heads
+                .keys()
+                .cloned()
+                .map(source::Branch)
+                .collect()
+        };
+
+        // TODO(finto): Can we do this by not going through string?
+        let hash = librad::hash::Hash::hash(remote.to_string().as_bytes());
+        // TODO(finto): This doesn't actually get the user. The peer id is their peer
+        // device rather than the Hash of their user profile.
+        let id = RadUrn::new(hash, Protocol::Git, Path::new());
```

Comment:
Need to get ahold of `<project>/remotes/<hash>/rad/self`. I think it's something like `storage.get_rad_self_of(project_urn, peer_id)`.

Need to mock peers to actually test this. Probably need to do some `git2` shenanigans.